src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolverEventSource.cs

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics.Tracing;
+using System.Threading.Tasks;
+using NuGet.Common;
+
+namespace Microsoft.Build.NuGetSdkResolver
+{
+    /// <summary>
+    /// Represents an <see cref="EventSource" /> for the NuGet-based MSBuild project SDK resolver.
+    /// </summary>
+    [EventSource(Name = "Microsoft-NuGet-SdkResolver")]
```

Comment:
Bear in mind for traces, Perf DDRITs, feedback bugs, targeted tracing etc to capture this, you'll need to opt them into this provider, or alternatively, reuse "Microsoft-VisualStudio-Common". The latter I'm less unclear of how that works to avoid clashes, etc.